utils/sl/comp_build.sh

```diff
@@ -0,0 +1,405 @@
+      for tarball in ${tarballs}; do
+        tarname=`basename ${tarball%_files.tar.gz}`
+        mkdir -p ${tarname}
+        pushd ${tarname}
+          tar -xzf ${tarball}
+        popd
+      done
+    popd
+  fi
+
+  echo "$name version is ${!upper_name}"
+}
+
+: ${JOB_SUFFIX:="-update-scratch"}
+if [ -d "/scratch/jenkins-2/artifacts" ];then
+  : ${CORAL_ARTIFACTS:="/scratch/jenkins-2/artifacts"}
+else
+  : ${CORAL_ARTIFACTS:="/scratch/coral/artifacts"}
+fi
+
+
+BUILD_CONFIG=`pwd`/build.config
+CUSTOM_SCRIPT=
+CUSTOM_SCRIPT_PASSED=
+
+while [ -n "$*" ]; do
+  case "$1" in
+    -c)
+      shift
+      BUILD_CONFIG=$(realpath $1)
+      shift
+      ;;
+    *)
+      CUSTOM_SCRIPT=$(realpath $1)
+      CUSTOM_SCRIPT_PASSED=${1}
+      shift
+      ;;
+  esac
+done
+
+if [ ! -f $BUILD_CONFIG ]; then
+  print_status "$BUILD_CONFIG was not found"
+  usage
+fi
+
+if [ -n "$CUSTOM_SCRIPT" ] && [ ! -f $CUSTOM_SCRIPT ]; then
+  print_status "<custom_build_script> $CUSTOM_SCRIPT was not found"
+  usage
+fi
+
+B_COMP=`cat $BUILD_CONFIG | grep "^component=" | sed 's/component=//'`
+DEPS=`cat $BUILD_CONFIG | grep "^depends=" | sed 's/depends=//'`
+
+if [ -z "$B_COMP" ]; then
+  print_status "component must be specified in $BUILD_CONFIG"
+  usage
+fi
+
+job_matrix=
+option=
+if [ -z "$WORKSPACE" ]; then
+  SET_PREFIX=
+  B_INS_PATH=`pwd`/install/`uname -s`
+  #Set job name so it thinks it's a review job
+  JOB_NAME=invalid_job_name
+  print_status "Not in Jenkins workspace"
+else
+  if [ "${DOCKER_IMAGE}x" == "x" ]; then
+    B_LINK_PATH="${CORAL_ARTIFACTS}/${JOB_NAME}/${BUILD_NUMBER}"
+    B_INS_PATH="${B_LINK_PATH}/${B_COMP}"
+  else
+    B_LINK_PATH=${DIST_TARGET}
+    B_INS_PATH=${B_LINK_PATH}/${B_COMP}
+  fi
+  SET_PREFIX="PREFIX=${B_INS_PATH}"
+  print_status "Building $B_INS_PATH in Jenkins workspace"
+  # This will probably break with folders.
+  # Folders are prepended to the displayed job name
+  # matrixes are appended to the displayed job name
+  # Not easy to detect if a matrix job is in a folder, so not trying for now.
+  is_matrix=${JOB_NAME#*/}
+  if [ "${is_matrix}" != "${JOB_NAME}" ]; then
+    # Not equal means a matrix
+    job_matrix="/${is_matrix}"
+  fi
+fi
+
+if [ -d "$CORAL_ARTIFACTS" ]; then
+  PREBUILT_AREA=
+  for dep in $DEPS; do
+    setup_dep $dep
+  done
+  option="PREBUILT_PREFIX=${PREBUILT_AREA}"
+fi
+
+set -x
+set -e
+rm -f ${B_COMP}-`uname -s`.conf
+if [ "${DOCKER_IMAGE}x" == "x" ]; then
+  scons $SET_PREFIX ${option}
+  scons install
+
+  if [ -n "$CUSTOM_SCRIPT" ]; then
+    #custom build step
+    print_status "Running custom build step"
+    COMP_PREFIX=${B_INS_PATH}
+    source $CUSTOM_SCRIPT
+  fi
+
+if [ -n "${B_LINK_PATH}" ]; then
+  ln -sfn ${BUILD_NUMBER} ${CORAL_ARTIFACTS}/${JOB_NAME}/latest
+fi
+else
+
+  docker_script=`find . -name docker_scons_comp_build.sh`
+
+  docker run --rm -u $USER \
+  --cap-add SYS_ADMIN --device /dev/fuse --privileged \
+  -v ${PWD}:/work -v ${WORK_TARGET}:${DIST_MOUNT} \
```

Comment:
(lint) Double quote to prevent globbing and word splitting. [SC2086]
(lint) Double quote to prevent globbing and word splitting. [SC2086]
(lint) Double quote to prevent globbing and word splitting. [SC2086]